In [1]:
import pandas as pd
from sqlalchemy import create_engine

In [2]:
db_url = "postgresql+psycopg2://postgres:abc123@localhost/prosperous_universe"
engine = create_engine(db_url)

In [3]:
query = """
    SELECT *
    FROM raw.fs_recipe_report;
"""

In [4]:
with engine.connect() as connection:
    df = pd.read_sql_query(query, con=connection)

In [5]:
df

,original_query,prefix,Output Material,input_materials_AI1_AskPrice_per_unit,total_input_materials_AI1_AskPrice,total_output_materials_AI1_BidPrice,profit_per_total,profit_per_day,traded_rank
0,FS:1xZR-1xAL=>1xAFR,FS,AFR,17089.500000,17089.500000,22000.0,4910.500000,8184.166667,11
1,FS:1xSTL-1xAL=>1xBFR,FS,BFR,4053.500000,4053.500000,7000.0,2946.500000,5893.000000,10
2,FS:1xAL-2xCU=>3xBRO,FS,BRO,1489.388889,4468.166667,6600.0,2131.833333,5329.583333,7
3,FS:4xAU-1xFE=>5xBGO,FS,BGO,2619.700000,13098.500000,15050.0,1951.500000,4878.750000,5
4,FS:1xCU-300xPE=>10xBCO,FS,BCO,556.933345,5569.333448,6910.0,1340.666552,4468.888507,2
5,FS:1xRGO-300xPE=>10xHCC,FS,HCC,610.000011,6100.000114,7250.0,1149.999886,3833.332952,6
6,FS:1xFE=>16xSFK,FS,SFK,82.406250,1318.500000,1440.0,121.500000,1012.500000,1
7,FS:1xBGO-300xPE=>10xBGC,FS,BGC,727.000011,7270.000114,7000.0,-270.000114,-900.000381,4
8,FS:1xSTL=>8xMFK,FS,MFK,395.500000,3164.000000,2408.0,-756.000000,-6300.000000,3
9,FS:2xSFK-6xFE=>1xUTS,FS,UTS,8145.000000,8145.000000,7000.0,-1145.000000,-2862.500000,8


In [7]:
# Use method='spearman' for ordinal rank data
corr = df['profit_per_day'].corr(df['traded_rank'], method='spearman')
print(f"Spearman correlation: {corr:.3f}")

Spearman correlation: -0.159


In [ ]:
# Use method='spearman' for ordinal rank data
corr = df['profit_per_day'].corr(df['traded_rank'], method='spearman')
print(f"Spearman correlation: {corr:.3f}")

Key Finding: A very weak negative Spearman correlation ($\rho = -0.159$) exists between traded_rank and profit_per_day across the 13 evaluated recipes. In practical terms, trade volume rank has almost no predictive relationship with daily profitability. High-volume items are just as likely to yield low or negative profits as lower-ranked niche items.